src/main/java/io/supertokens/oauth/OAuth.java

```diff
@@ -90,6 +102,132 @@ public static OAuthAuthResponse getAuthorizationUrl(Main main, AppIdentifier app
         return new OAuthAuthResponse(redirectTo, cookies);
     }
 
+    //This more or less acts as a pass-through for the sdks, apart from camelCase <-> snake_case key transformation and setting a few default values
+    public static JsonObject registerOAuthClient(Main main, AppIdentifier appIdentifier, Storage storage, JsonObject paramsFromSdk)
+            throws TenantOrAppNotFoundException, InvalidConfigException, IOException, OAuthClientRegisterException,
+            NoSuchAlgorithmException, StorageQueryException {
+
+        OAuthStorage oauthStorage = StorageUtils.getOAuthStorage(storage);
+        String adminOAuthProviderServiceUrl = Config.getConfig(appIdentifier.getAsPublicTenantIdentifier(), main).getOAuthProviderAdminServiceUrl();
+
+        byte[] idBaseBytes = new byte[48];
+
+        while(true){
+            new SecureRandom().nextBytes(idBaseBytes);
+            String clientId = "supertokens_" + Utils.hashSHA256Base64UrlSafe(idBaseBytes);
+            try {
+                if(oauthStorage.isClientIdAlreadyExists(clientId)){
+                    continue; // restart
+                }
```

Comment:
this check doesn't really help cause we can have multiple cores (with their own db) connected to one instance of hydra. Also, i assume that if we try and create a new client id with an existing one, hydra itself will throw an error? So we should detect this case based on what hydra responds with

Reply:
Yeah, I didn't think about the multiple cores to the same hydra scenario. But this at least can guarantee that we don't try to register a new client with already existing clientId of which we know.
About the multiple cores to the same hydra instance, should we silently retry if hydra response is something like "Client already exists?"
(In that case hydra returns `409 Conflict` with a body of `{
    "error": "Unable to insert or update resource because a resource with that value exists already",
    "error_description": ""
}` )

Reply:
Yup. So if hydra already tells us about this issue, i don't see why we need to also check for it